zerver/views/__init__.py

```diff
@@ -764,7 +773,7 @@ def create_realm(request, creation_key=None):
                               request=request)
 
 def confirmation_key(request):
-    return json_success(request.session.get('confirmation_key'))
+    return json_success({'confirmation_key': os.getenv('CONFIRMATION_KEY')})
```

Comment:
When using subdomains `session.get('confirmation_key')` was returning Null. I was not able to  debug that.